src/EditorFeatures/Core/GoToDefinition/GoToDefinitionCommandHandler.cs

```diff
@@ -161,32 +163,30 @@ private static void ReportFailure(Document document)
         {
             bool succeeded;
 
-            var indicatorFactory = document.Project.Solution.Services.GetRequiredService<IBackgroundWorkIndicatorFactory>();
-
             // TODO: prior logic was to get a tracking span of length 1 here.  Preserving that, though it's unclear if
             // that is necessary for the BWI to work properly.
             Contract.ThrowIfTrue(position.Snapshot.Length == 0);
             var applicableToSpan = position < position.Snapshot.Length
                 ? new SnapshotSpan(position, position + 1)
                 : new SnapshotSpan(position - 1, position);
 
-            using (var backgroundIndicator = indicatorFactory.Create(
+            using (var backgroundIndicator = _backgroundWorkIndicatorService.Create(
                 args.TextView, applicableToSpan,
                 EditorFeaturesResources.Navigating_to_definition))
             {
-                var cancellationToken = backgroundIndicator.UserCancellationToken;
+                var cancellationToken = backgroundIndicator.CancellationToken;
 
                 // determine the location first.
                 var (location, _) = await service.FindDefinitionLocationAsync(
                     document, position, includeType: true, cancellationToken).ConfigureAwait(false);
 
                 // make sure that if our background indicator got canceled, that we do not still perform the navigation.
-                if (backgroundIndicator.UserCancellationToken.IsCancellationRequested)
+                if (backgroundIndicator.CancellationToken.IsCancellationRequested)
                     return;
 
                 // we're about to navigate.  so disable cancellation on focus-lost in our indicator so we don't end up
                 // causing ourselves to self-cancel.
-                backgroundIndicator.CancelOnFocusLost = false;
```

Comment:
Does `CancelOnFocusLost` default to true or false? I see cases above that explicitly pass `CancelOnFocusLost = true`, yet here we did not explicitly state that and are now setting `CancelOnFocusLost = false`. It seems like one of the two is redundant.

Reply:
I'd have to check implementation, but I'm not opposed to explicitly setting it on every creation. My guess it defaults to false because they are `{ get; init; }` defined. @kayle ?